### PR TITLE
Add support for gitlab cli

### PR DESCRIPTION
**Pull request by AWildErin**
I've added support for GitLab's CLI program. Currently, it'll base it off the git URL and default gitlab's executable if the URL isn't either GitLab or GitHub. The reasoning behind this choice is that the chances of it being a GitLab repo on a self-hosted instance are higher than on a self-hosted GitHub instance.

I also turned off the prompt for when we run --build source because it's likely going to be implied you've already set up the project, and that function sort of makes you download UE4 despite the fact the later code finds my EGS install of Unreal Engine 4 just fine. That commit can probably be dropped, however, but I don't see much harm in it personally.

GitHub support *shouldn't* have broke, but I don't have a repo set up to make sure this is the case, but logically it should still work fine.

### CODE REVIEW

File: pbpy/pbunreal.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def get_bundle_verification_file(bundle_name):
 
 @lru_cache()
 def get_engine_base_path():
-    root = get_engine_install_root()
+    root = get_engine_install_root(False)
```
Comment (mastercoms) [inline pbpy/pbunreal.py]: This breaks a lot of user flows. Your CI machine should define CI=1, and that _should_ make the `get_engine_install_root` to not prompt.

**Review by mastercoms — CHANGES_REQUESTED** [review]

File: pbpy/pbunreal.py (hunk 1 above)
Comment (AWildErin) [reply pbpy/pbunreal.py]: Ahh, that makes sense. I'll revert this change!

Comment (AWildErin) [reply pbpy/pbunreal.py]: I've dropped the commit. Though I did run into an issue where, once CI=1 is set, it would *only* look for UE4 inside `C:\ue4`.

Comment (mastercoms) [reply pbpy/pbunreal.py]: Refer to .ue4v-user-ci to see how environment variables can be set for CI for configuration

Comment (AWildErin) [reply pbpy/pbunreal.py]: Okay I've sort of gotten it working. It's failing in `get_engine_base_path` as it's an EGS install. I've added a function that checks for `.egstore` similar to `is_source_install`. Do you mind if this is added? On the build machines I have, this works fine as we don't have any specific requirements that would require us to build the engine ourselves, plus PBSync works fine with marketplace builds.

Comment (mastercoms) [reply pbpy/pbunreal.py]: But get_engine_base_path already has handling for installed builds, are you saying this behavior no longer works?

https://github.com/ProjectBorealis/PBSync/blob/4c65e7d99bf2570550dfe86868749725aeb4d11e/pbpy/pbunreal.py#L590-L595

Comment (AWildErin) [reply pbpy/pbunreal.py]: It doesn't work, as that code is run when if root is none, which in my case it isn't. As it doesn't fulfil the other checks below, `get_engine_base_path` returns none.

Comment (mastercoms) [reply pbpy/pbunreal.py]: Why isn't root none if you aren't using a source install? You shouldn't need to specify the env or any path variable at all if you are using an EGS install.

Comment (AWildErin) [reply pbpy/pbunreal.py]: `get_engine_install_root` checks for the dirs in the config, it gets them fine and returns that path, so `root` is set correctly to the path I supplied to it

**Review by mastercoms — APPROVED** [review]

**Pull request merged**